server/route.go

```diff
@@ -21,70 +22,128 @@ var (
 	errTimeout    = errors.New("Channel sending timeout")
 )
 
+type RouteOptions struct {
+	Path protocol.Path
+
+	// channel buffer size
+	Size int
+
+	// QueueSize
+	QueueSize int
+
+	// Timeout to define how long to wait for the message to be read on the channel
+	// if timeout is reached the route is closed
+	Timeout time.Duration
+}
+
+type RouteParams map[string]string
```

Comment:
RouteParams and all associated func(s) to could be moved to a separate file, route_params.go ?
(also the corresponding tests)
=> route.go will become shorter and more focused